patterns/observers/observer-delegate.cs

```diff
@@ -0,0 +1,82 @@
+// Generic Delegate Implementation of the Observer Pattern
+
+/*
+References used: 
+https://www.youtube.com/watch?v=yoAOo4dM2pk
+https://www.c-sharpcorner.com/article/design-pattern-with-net-delegates/
+https://www.devleader.ca/2023/11/17/examples-of-the-observer-pattern-in-c-how-to-simplify-event-management/#google_vignette
+*/
+
+using System;
+
+// Define the generic delegate for the event handler
+public delegate void EventHandler<T>(T data);
+
+// Define the Observer interface
+public interface IObserver<T>
+{
+    void Update(T data);
+}
+
+// Define the Subject class
+public class Subject<T>
+{
+    // Event 'DataChanged' to which observers can subscribe
+    public event EventHandler<T> DataChanged;
+
+    // Method to notify observers when state of the subject changes
+    public void Notify(T data)
+    {
+        DataChanged?.Invoke(data);
+    }
+}
```

Comment:
As @abhiram-shaji suggested - I think maintaining a list of subscribers here is good Observer pattern practice in your Subject implementation.

Reply:
@ash-teach I have added methods to subscribe/unsubscribe an observer in the Subject class in my latest commit. Please review.